### PR TITLE
Fix bug 7197 and make certifiable

### PR DESCRIPTION
**Pull request by misslivirose**
An update to the emoji app.
This is the base version of the app that needs to be re-uploaded to marketplace. Please disregard the absolute URLs in JSON - these will reference the final versions of the assets when uploaded to marketplace. 

QA will run on the marketplace uploaded version and is not needed for this PR separately.

### CODE REVIEW

File: Marketplace/emoji-tablet/JSON_files/pickle.json [hunk 1]
```diff
@@ -28,6 +28,6 @@
     "lifetime":30,
     "grabbable":true,
     "collisionless":false,
-    "script":"https://hifi-content.s3.amazonaws.com/elisalj/emoji_scripts/behaviors/crunch.js",
+    "script":"file:///D:/hifi-content/Marketplace/emoji-tablet/emoji_scripts/behaviors/crunch.js",
```
Comment (thoys) [inline Marketplace/emoji-tablet/JSON_files/pickle.json]: local path?

File: Marketplace/emoji-tablet/JSON_files/pizza.json [hunk 2]
```diff
@@ -28,6 +28,6 @@
     "lifetime":30,
     "grabbable":true,
     "collisionless":false,
-    "script":"https://hifi-content.s3.amazonaws.com/elisalj/emoji_scripts/behaviors/Edible.js",
+    "script":"file:///D:/hifi-content/Marketplace/emoji-tablet/emoji_scripts/behaviors/Edible.js",
```
Comment (thoys) [inline Marketplace/emoji-tablet/JSON_files/pizza.json]: local path?

File: Marketplace/emoji-tablet/JSON_files/pickle.json (hunk 1 above)
Comment (elisa-lj11) [inline Marketplace/emoji-tablet/JSON_files/pickle.json]: Can you use `Script.resolvePath("../emoji_scripts/behaviors/crunch.js")` ?

File: Marketplace/emoji-tablet/JSON_files/pizza.json (hunk 2 above)
Comment (elisa-lj11) [inline Marketplace/emoji-tablet/JSON_files/pizza.json]: Can you use `Script.resolvePath()` here too?

File: Marketplace/emoji-tablet/behaviors/Edible.js [hunk 3]
```diff
@@ -29,7 +29,8 @@
     var checkIfNearHead = function () {
         var position = Entities.getEntityProperties(_this.entityID, 'position').position;
         var avatarHeadPosition = MyAvatar.getJointPosition("Head");
-        if (isWithinDistance(position.y, avatarHeadPosition.y) &
+        if (isWithinDistance(position.x, avatarHeadPosition.x) &
+            isWithinDistance(position.y, avatarHeadPosition.y) &
```
Comment (elisa-lj11) [inline Marketplace/emoji-tablet/behaviors/Edible.js]: You could use `Vec3.distance(position, MyAvatar.getJointPosition("Head")) < DISTANCE_WITHIN` to cut down on lines

File: Marketplace/emoji-tablet/emojiTablet.js [hunk 4]
```diff
@@ -12,17 +12,18 @@
...
     var APP_NAME = "EMOJIS";
-    var APP_URL = "https://hifi-content.s3.amazonaws.com/elisalj/emoji_scripts/emojiTabletUI.html?" + Date.now();
+    var APP_URL = Script.resolvePath("emojiTabletUI.html");
     var APP_ICON = "./icons/emoji-i.svg";
     var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
+    var isOpen = false;
 
     var DISTANCE = 0.5;
-    var WANT_DEBUG = false;
+    var WANT_DEBUG = true;
```
Comment (elisa-lj11) [inline Marketplace/emoji-tablet/emojiTablet.js]: should this still be `true`?

File: Marketplace/emoji-tablet/emojiTablet.js [hunk 5]
```diff
@@ -31,17 +32,43 @@ var library = Script.require("https://hifi-content.s3.amazonaws.com/elisalj/emoj
...
+                x: 0.0,
+                y: 0.4,
+                z: -0.5
+            };
+
+        } else {
+            rezOffset = {
+                x: 0.0,
+                y: 0.7,
+                z: -0.5
```
Comment (elisa-lj11) [inline Marketplace/emoji-tablet/emojiTablet.js]: yaay im glad this is getting used

File: Marketplace/emoji-tablet/JSON_files/pickle.json (hunk 1 above)
Comment (misslivirose) [reply Marketplace/emoji-tablet/JSON_files/pickle.json]: No, this is a JSON file and the comment explicitly calls out that these aren't final

File: Marketplace/emoji-tablet/JSON_files/pizza.json (hunk 2 above)
Comment (misslivirose) [reply Marketplace/emoji-tablet/JSON_files/pizza.json]: No, this is a JSON file and the comment explicitly calls out that these aren't final

File: Marketplace/emoji-tablet/JSON_files/pickle.json (hunk 1 above)
Comment (elisa-lj11) [reply Marketplace/emoji-tablet/JSON_files/pickle.json]: oops, I didn't see the comment at the top of this PR, sorry about that

File: Marketplace/emoji-tablet/behaviors/Edible.js [hunk 6]
```diff
@@ -29,9 +29,7 @@
     var checkIfNearHead = function () {
         var position = Entities.getEntityProperties(_this.entityID, 'position').position;
         var avatarHeadPosition = MyAvatar.getJointPosition("Head");
-        if (isWithinDistance(position.x, avatarHeadPosition.x) &
-            isWithinDistance(position.y, avatarHeadPosition.y) &
-            isWithinDistance(position.z, avatarHeadPosition.z)) {
+        if (Vec3.distance(position, avatarHeadPosition) >= DISTANCE_WITHIN) {
```
Comment (elisa-lj11) [inline Marketplace/emoji-tablet/behaviors/Edible.js]: should this be `<=` ?

Comment (misslivirose) [reply Marketplace/emoji-tablet/behaviors/Edible.js]: Ack yes it should, nice catch

**Pull request closed without merging**